File: app/src/main/java/toong/vn/androidroom/data/local/AppDatabase.java
package toong.vn.androidroom.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import toong.vn.androidroom.data.local.dao.TaskDao;
import toong.vn.androidroom.data.local.dao.UserDao;
import toong.vn.androidroom.data.model.Task;
import toong.vn.androidroom.data.model.User;

@Database(entities = { User.class, Task.class }, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract UserDao userDao();
    public abstract TaskDao mTaskDao();

    private static AppDatabase INSTANCE;

    public synchronized static AppDatabase getAppDatabase(Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE =
                            Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class,
                                    "android-room-user-database").build();
                }
            }
        }
        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }
}

File: app/src/main/java/toong/vn/androidroom/data/local/AppDatabase.kt
package toong.vn.androidroom.data.local

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context
import toong.vn.androidroom.data.local.dao.TaskDao
import toong.vn.androidroom.data.local.dao.UserDao
import toong.vn.androidroom.data.model.Task
import toong.vn.androidroom.data.model.User

@Database(entities = [User::class, Task::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun userDao(): UserDao
    abstract fun taskDao(): TaskDao

    companion object {
        fun getAppDatabase(context: Context): AppDatabase {
            return Room.databaseBuilder(
                    context,
                    AppDatabase::class.java, "database-name"
            ).build()
        }
    }
}